src/components/globalSearchBox/globalSearchBox.component.tsx

```diff
@@ -1,25 +1,98 @@
+"use client";
+
+import { FormEvent, useContext, useEffect, useState } from "react";
+
+import { usePathname, useRouter } from "next/navigation";
+
 import MingcuteSearchFill from "@/icon/MingcuteSearchFill";
 import MingcuteLocationFill from "@/icon/MingcuteLocationFill";
 
+import { FiltersContext } from "@/app/search/providers/filters/filters.provider";
+
 import styles from "./globalSearchBox.module.css";
 
 export default function GlobalSearchBoxComponent() {
+  const router = useRouter();
+  const pathname = usePathname();
+
+  const { filters, dispatchFilters } = useContext(FiltersContext);
+
+  const [query, setQuery] = useState<string>("");
+
+  const formSubmitHandler = (e: FormEvent<HTMLFormElement>): void => {
+    e.preventDefault();
+
+    if (pathname === "/search") {
+      if (query) {
+        dispatchFilters({
+          type: "updated_filter",
+          key: "query",
+          value: query,
+        });
+      }
+    } else {
+      const href = query ? `/search/?query=${query}` : "/search";
+      router.push(href);
+    }
+  };
+
+  useEffect(() => {
+    if (pathname !== "/search") {
+      return;
+    }
+
+    const filterQuery = filters.query || "";
+    setQuery(filterQuery);
+
+    const filterExpertise = filters.expertise;
+    const filterGender = filters.gender;
+    const filterIsVerified = filters.isVerified;
+    const filterOrdering = filters.ordering;
+
+    const searchParams = [];
```

Comment:
زمانی که مقدار اولیه، آرایه‌ی خالیه... بهتره تایپ رو مشخص کنید
```suggestion
    const searchParams: string[] = [];
```